lib/jobs/upsert-node-job.js

```diff
@@ -0,0 +1,41 @@
+// Copyright 2016, EMC, Inc.
+/* jshint node:true */
+
+'use strict';
+
+module.exports = UpsertNodeAttributeFactory;
+
+var di = require('di');
+
+di.annotate(UpsertNodeAttributeFactory, new di.Provide('Job.Upsert.Node.Attribute'));
+di.annotate(UpsertNodeAttributeFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Services.Waterline',
+    'Logger'
+    )
+);
+
+function UpsertNodeAttributeFactory(BaseJob, util, waterline, Logger) {
+    var logger = Logger.initialize(UpsertNodeAttributeFactory);
+
+    function UpsertNodeAttribute(options, context, taskId) {
+        UpsertNodeAttribute.super_.call(this, logger, options, context, taskId);
+        this.nodeId = this.options.target;
```

Comment:
A suggestion: The context will still pass a `target` to each job ( see https://github.com/RackHD/on-http/blob/master/lib/services/workflow-api-service.js#L77), I think the options.target can be default to `context.target` if no `options.target`, of course this change need remove the `target` from requiredOptions list in base-task.